cpp/src/strings/regex/regcomp.cpp

```diff
@@ -53,6 +54,13 @@ static reclass ccls_S(16); // not ccls_s
 static reclass ccls_d(4);  // digits [0-9]
 static reclass ccls_D(32); // not ccls_d plus '\n'
 
+// Tables for analyzing quantifiers
+std::set<int> valid_preceding_inst_types( {CHAR, CCLASS, NCCLASS, ANY, ANYNL} );
```

Comment:
I wouldn't use `std::set` for fixed-size arrays -- paying a lot of overhead. These are so small that you would probably be better off using `std::array` (since size is known at compile time), and then use `std::find()` to search them.  (You could also pre-sort thew contents and use `std::binary_search()`).